k8s/k8s.go

```diff
@@ -388,7 +377,36 @@ func CmdDelK8s(ctx context.Context, c calicoclient.Interface, epIDs utils.WEPIde
 	}
 
 	logger.Info("Teardown processing complete.")
+	return nil
+}
 
+// releaseIPAddrs calls directly into Calico IPAM to release the IP addresses in the given annotation string.
+// NOTE: This function assumes Calico IPAM is in use, and calls into it directly rather than calling the IPAM plugin.
+func releaseIPAddrs(ipAddrs string, calico calicoclient.Interface, logger *logrus.Entry) error {
+	logger.Infof("Parsing annotation \"cni.projectcalico.org/ipAddrs\":%s", ipAddrs)
+
+	// We need to make sure there is only one IPv4 and/or one IPv6
+	// passed in, since CNI spec only supports one of each right now.
+	ipList, err := validateAndExtractIPs(ipAddrs, "cni.projectcalico.org/ipAddrs", logger)
+	if err != nil {
+		return err
+	}
+
+	// For each IP, call out to Calico IPAM to release it.
+	for _, ip := range ipList {
+		log := logger.WithField("IP", ip)
+		log.Info("Releasing explicitly requested address")
+		unallocated, err := calico.IPAM().ReleaseIPs(context.Background(), []cnet.IP{{ip}})
```

Comment:
wondering if you can avoid this loop if you just did `ReleaseIPs(ctx, []cnet.IP{ipList...}` something like that? since ReleaseIPs can release multiple IPs

Reply:
that's what the `...` is for, it'll expand the list. Haven't tried it, but it's ok if that doesn't work

Reply:
But the contents of ipList aren't `cnet.IP` objects, they're `net.IP` objects, so we'd need to do something like 

`[]cnet.IP{cnet.IP{ip} for net in ipList}`, so we're iterating over the list anyway.